### PR TITLE
coins/constants: New regtest functionality

### PR DESCRIPTION
**Pull request by dev-warrior777**

### CODE REVIEW

**Comment by dev-warrior777** [comment]
_Testing:_
**Coins.py** main
[coins.py.txt](https://github.com/firoorg/electrum-firo/files/11419996/coins.py.txt)

**electrum-firo-env**
```
. . .
if [ -z "${REGTEST}" ]
then
    echo 'normal startup mainnet'
    ./electrum-firo "$@"
else
    echo "startup on the electrumX regtest network with one tcpip server at 127.0.0.1:50001:t"
    ./electrum-firo -v --regtest --oneserver --server 127.0.0.1:50001:t
fi
```

**Comment by dev-warrior777** [comment]
image

May I know which test failed please?

Is it logic, linting ?

File: electrum_dash/coins.py [hunk 1]
```diff
@@ -16,6 +16,9 @@ class Firo(Coin):
 
     @classmethod
     def static_header_offset(cls, height):
+        if (cls.__name__== 'FiroRegtest'):
+            return cls.PRE_MTP_HEADER_SIZE * height
```
Comment (psolstice) [inline electrum_dash/coins.py:20]: Don't we have the same effect by setting `PRE_MTP_BLOCKS` to `INT_MAX`? These ifs aren't looking great.

**Review by psolstice — CHANGES_REQUESTED** [review]

File: electrum_dash/coins.py [hunk 2]
```diff
@@ -64,3 +73,10 @@ class FiroTestnet(Firo):
     PRE_MTP_BLOCKS = 1
     AFTER_PROGPOW_BLOCK_HEIGHT = 37305
     PROGPOW_START_TIME = 1630069200
+
+class FiroRegtest(Firo):
+    PRE_MTP_BLOCKS = 2147483647             # INT_MAX 0xefffffff
```
Comment (psolstice) [inline electrum_dash/coins.py:78]: also why `0xefffffff`? `INT_MAX` is `0x7fffffff`

Comment (dev-warrior777) [reply electrum_dash/coins.py:78]: Thanks for pointing this out
image

The comment is wrong the number is correct. msb not set. Will change the comment also. My tests so far showing no need for the short-circuit logic above but this time am testing `0x7fffffff` cases.

Comment (dev-warrior777) [reply electrum_dash/coins.py:78]: There is one minor edge case in:
https://github.com/firoorg/electrum-firo/blob/97873167768250e60ade56a17f717a0125e2ba97/electrum_dash/coins.py#L35

if `height` is 2147483647 (INT_MAX) then 120 is the returned size of the header. 
It is regtest and never going to hit that height

So I will clean up my code and make a new commit with the short-circuit code removed and updated comments

Comment (dev-warrior777) [reply electrum_dash/coins.py]: Fixed in: https://github.com/firoorg/electrum-firo/pull/41/commits/0fdea46083aff75ab30ab5499c59a537c11db6de

**Review by psolstice — APPROVED** [review]

**Comment by dev-warrior777** [comment]
Thank you
